podman_tutorial.md

```diff
@@ -0,0 +1,156 @@
+# cd ~/src/github.com/containernetworking/plugins
+# ./build
+# sudo mkdir -p /usr/libexec/cni
+# sudo cp bin/* /usr/libexec/cni
+# git clone https://github.com/opencontainers/runc.git ~/src/github.com/opencontainers/runc
+# GOPATH=~/ make static BUILDTAGS="seccomp selinux"
+# sudo cp runc /usr/bin/runc
+# sudo mkdir -p /usr/local/libexec
+# sudo ln -s /usr/libexec/crio /usr/local/libexec/
+```
+
+) -->
+## Familiarizing yourself with Podman
+
+### Running a sample container
+This sample container will run a very basic httpd server that serves only its index
+page.
+```
+# sudo podman run -dt -e HTTPD_VAR_RUN=/var/run/httpd -e HTTPD_MAIN_CONF_D_PATH=/etc/httpd/conf.d \
+                    -e HTTPD_MAIN_CONF_PATH=/etc/httpd/conf \
+                    -e HTTPD_CONTAINER_SCRIPTS_PATH=/usr/share/container-scripts/httpd/ \
+                    registry.fedoraproject.org/f26/httpd /usr/bin/run-httpd
+```
+Because the container is being run in detached mode, represented by the *-d* in the podman run command, podman
+will print the container ID after it has run.
+
+### Listing running containers
+The Podman *ps* command is used to list creating and running containers.
+```
+# sudo podman ps
+```
+
+Note: If you add *-a* to the *ps* command, Podman will show all containers.
+
+### Executing a command in a running container
+You can use the *exec* subcommand to execute a command in a running container.  Eventually you will be able to
+obtain the IP address of the container through inspection, but that is not enabled yet.  Therefore, we will
+install *iproute* in the container.  Notice here that we use the switch **--latest** as a shortcut for the latest
+created container.  You could also use the container's ID listed during *podman ps* in the previous step or
+when you ran the container.
+```
+# sudo podman exec --latest -t dnf -y install iproute
+# sudo podman exec --latest -t ip a
+```
+
+Note the IP address of the *eth0* device.
+
+### Testing the httpd server
+Now that we have the IP address of the container, we can test the network communication between the host
+operating system and the container using curl. The following command should display the index page of our
+containerized httpd server.
+```
+# curl http://<IP_address>:8080
+```
+
+### Viewing the container's logs
+You can view the container's logs with Podman as well:
+```
+# sudo podman logs --latest
+```
+
+<!-- (
+### Viewing the container's pids
+And you can observe the httpd pid in the container with *top*.
+```
+# sudo podman top <container_id>
+``` ) -->
+### Stopping the container
+To stop the httpd container:
+```
+# sudo podman stop --latest
+```
+You can also check the status of one or more containers using the *ps* subcommand. In this case, we should
+use the *-a* argument to list all containers.
+```
+# sudo podman ps -a
+```
+
+### Removing the container
+To remove the httpd container:
+```
+# sudo podman rm --latest
+```
+
```

Comment:
your call.  would another "sudo podman ps -a" here to show the container is gone be worth while?

Reply:
done